app/queries/filter_rewards.rb

```diff
@@ -0,0 +1,17 @@
+class FilterRewards
+  attr_accessor :initial_scope
+
+  def initialize(initial_scope)
```

Comment:
It looks like you are passing `Reward.order(created_at: :desc)` which actually does not depends on any params sent to the controller, so I would suggest using it inside the query object, also instead you should pass params here

```suggestion
  def initialize(initial_scope)
```